web/src/components/IdeViewer/IdeViewer.js

```diff
@@ -74,43 +87,43 @@ let currentCode // I have no idea why this works and using state.code is the dis
 const IdeViewer = () => {
   const { state, dispatch } = useContext(IdeContext)
   const [isDragging, setIsDragging] = useState(false)
+  const [image, setImage] = useState()
 
-  const image = useMemo(
-    () =>
+  useEffect(() => {
+    setImage(
       state.objectData?.type === 'png' &&
-      state.objectData?.data &&
-      window.URL.createObjectURL(state.objectData?.data),
-    [state.objectData]
-  )
+        state.objectData?.data &&
+        window.URL.createObjectURL(state.objectData?.data)
+    )
+    setIsDragging(false)
+  }, [state.objectData])
   currentCode = state.code
   return (
     <div className="p-8 border-2 m-2">
-      <div className="pb-4">
-        Rotating the Three.js Cube should than update the openscad camera to
-        view the CAD object from the same angle. But having trouble translating
-        between the two coordinate systems. OpenScad's camera only changes X and
-        Z angles and Y is always 0. Where as rotation values from Three seem to
-        change all x, y and z. I probably need to learn a bit more about 3d
-        math.
-      </div>
       <div className="relative" style={{ height: '500px', width: '500px' }}>
+        {state.isLoading && (
+          <div className="inset-0 absolute flex items-center justify-center">
+            <div className="h-16 w-16 bg-pink-600 rounded-full animate-ping"></div>
+          </div>
+        )}
         {image && (
           <div
-            className="absolute inset-0"
-            style={{ opacity: isDragging ? '0%' : '100%' }}
+            className={`absolute inset-0 transition-opacity duration-500 ${
+              isDragging ? 'opacity-25' : 'opacity-100'
+            }`}
           >
             <img src={image} className="" />
           </div>
         )}
         <div
-          className={`opacity-0 absolute inset-0 ${
+          className={`opacity-0 opacity- absolute inset-0 transition-opacity duration-500 ${
```

Comment:
It looks like you don't need this "opacity-" right?

Reply:
Oh yup 100% thanks for pointing this out.